src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/TypeSymbolExtensions.cs

```diff
@@ -277,30 +277,24 @@ public static INamedTypeSymbol ResolveUnboundConstructedTypeToConstructedType(th
             extraTypeArgumentsInTemplate = typeArgumentsToSubstitute.Length - currentArityOffset;
 
             return currentType;
+        }
 
-            // Get the type arguments and their nullable annotations from the instantiated generic type, including all levels of nesting.
-            // Examples:
-            // Foo<A> -> [A]
-            // Foo.Bar<A> -> [A]
-            // Foo<A?>.Bar<B> -> [A?, B]
-            // Foo<A, B, C?>.Bar<D?, E> -> [A, B, C?, D?, E]
-            static (ImmutableArray<ITypeSymbol>, ImmutableArray<NullableAnnotation>) GetArgumentsToSubstitute(INamedTypeSymbol instantiatedGeneric)
+        public static (ImmutableArray<ITypeSymbol> TypeArguments, ImmutableArray<NullableAnnotation> TypeArgumentNullableAnnotations) GetAllTypeArgumentsIncludingInContainingTypes(this INamedTypeSymbol genericType)
```

Comment:
Please comment and fully document what this method is doing and how. We have two loops and populating in a way that would seem order is important.